src/shared/containers/GetOrganizations.tsx

```diff
@@ -45,28 +46,51 @@ const canAccessCheckout = (me: Me): boolean => {
 
 const GetOrganizations: FunctionComponent = () => {
   const {status, org} = useSelector(getAllOrgs)
+
   const quartzMeStatus = useSelector(
     (state: AppState) => state.me.quartzMeStatus
   )
   const quartzMe = useSelector(getQuartzMe)
+
+  const quartzIdentityStatus = useSelector(
+    (state: AppState) => state.identity.status
+  )
+
   const {id: meId = '', name: email = ''} = useSelector(getMe)
   const dispatch = useDispatch()
 
+  // Ecommerce is refactoring so that we have a flag similar to isRegionBeta coming from the /quartz/identity endpoint.
+  useEffect(() => {
+    // Remove quartzIdentity condition once flag is deployed for all users.
+    if (
+      CLOUD &&
+      isFlagEnabled('quartzIdentity') &&
+      isFlagEnabled('uiUnificationFlag') &&
+      quartzMe?.isRegionBeta === null
+    ) {
+      dispatch(getCurrentOrgDetailsThunk())
+    }
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps
+
+  // This doesn't require another API call.
   useEffect(() => {
     if (status === RemoteDataState.NotStarted) {
       dispatch(getOrganizations())
     }
-  }, [dispatch, status])
+  }, [dispatch, status]) // eslint-disable-line react-hooks/exhaustive-deps
 
   useEffect(() => {
     if (
       isFlagEnabled('uiUnificationFlag') &&
       quartzMeStatus === RemoteDataState.NotStarted
+      // For now, just check whether quartzMeStatus is not set, because quartzMe is what is
+      // currently being used by the application.
```

Comment:
This is another one of those comments that has a tendency to get out of date and not be removed which will cause confusion down the road

Reply:
Removed this and some other comments that could end up being out of date.